src/util/stream-to-promise.js

```diff
@@ -0,0 +1,9 @@
+/* @flow */
+
+export async function streamToPromise(
```

Comment:
[stream-to-promise](https://github.com/bendrucker/stream-to-promise) depends on [end-of-stream](https://github.com/mafintosh/end-of-stream), which [calls the callback on `finish` event](https://github.com/mafintosh/end-of-stream/blob/master/index.js#L64).

[From Node.js document](https://nodejs.org/api/stream.html#stream_event_finish),

> The 'finish' event is emitted after the stream.end() method has been called, and all data has been flushed to the underlying system.

If we need to wait for the release of the file descriptor, we should listen to [`close` event](https://nodejs.org/api/stream.html#stream_event_close).

> The 'close' event is emitted when the stream and any of its underlying resources (a file descriptor, for example) have been closed.

So the problem is: when the promise is resolved, the file is still open, which make tmp module fail to do the cleanup task and leave some junk data in the file system.

Related issue: https://github.com/mozilla/web-ext/issues/239